### PR TITLE
Add support for flushing headers

### PR DESCRIPTION
**Pull request by PlasmaPower**
This is a breaking change, as seen by the test change, but it is very unlikely to break things. Previous discussion was at #668. Any more ideas for tests?

### CODE REVIEW

**Comment by PlasmaPower** [comment]
Branch coverage dropped a branch from 218 to 217, but statement coverage remained the same. However, codecov didn't seem to process this PR, so I don't know which branch it is.

**Comment by jonathanong** [comment]
ah crap... because i need to add codecov to v2.x as well. damnit

**Comment by PlasmaPower** [comment]
I thought that might be the case. Also, coveralls still seems to be running, I'm assuming we want to remove that?

File: test/response/flushHeaders.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      }
+      try {
+        ctx.status = 200;
+      } catch (err) {
+        ctx.body += 'ctx.status fail ';
+      }
+      try {
+        ctx.length = 10;
+      } catch (err) {
+        ctx.body += 'ctx.length fail';
```
Comment (jonathanong) [inline test/response/flushHeaders.js]: interesting way to make sure things are asserted... haha

**Pull request merged**